pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -231,58 +255,134 @@ private DataSchema getPrePostAggregationDataSchema(DataSchema dataSchema) {
     return new DataSchema(columnNames, columnDataTypes);
   }
 
-  private IndexedTable getIndexedTable(DataSchema dataSchema, Collection<DataTable> dataTables) {
+  private IndexedTable getIndexedTable(DataSchema dataSchema, Collection<DataTable> dataTablesToReduce,
+      DataTableReducerContext reducerContext)
+      throws TimeoutException {
+    long start = System.currentTimeMillis();
+    int numDataTables = dataTablesToReduce.size();
```

Comment:
We can short circuit the single data table case by directly return the `new SimpleIndexedTable(dataSchema, _queryContext, capacity)`

Reply:
I did that initially. But then it requires indexTable.finish(). And in future anytime we have other such tasks that need to be done before returning, they will need to be done at two places. So I chose to avoid that.